File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
n = int(input())

# 3, 4, 5, 6, 7, 8, 9, 10, 11, 12
result = [1, -1, 1, 2, -1, 2, 3, 2, 3, 4]

count = 0

while n > 12:
  n -= 5
  count += 1

count += result[n-3]
print(count)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.cc
#include <stdio.h>
#pragma warning(disable : 4996)

int main(void) {
	int n;

	scanf("%d", &n);

	int data[10] = { 1, -1, 1, 2, -1, 2, 3, 2, 3, 4 };

	int result = 0;

	while (n > 12) {
		n -= 5;
		result += 1;
	}

	result += data[n - 3];

	printf("%d\n", result);
	return 0;
}